File: mqxq-front/src/api/types/course.ts
// 基础课程信息
export interface Course {
  id: number
  teacherId: number
  subjectId: number
  title: string
  description: string
  price: number
  lessonNum: number
  durationSum: number
  cover: string
  buyCount: number
  viewCount: number
  status: number
  publishTime: string
  createTime: string
  createUser: number
  updateTime: string
  updateUser: number
}

// 我的课程
export interface MyCourse {
  id: number
  userId: number
  courseId: number
  orderId: number
  status: number
  createTime: string
  updateTime: string
  // 关联的课程信息
  course?: Course
}

// 课程评论
export interface CourseComment {
  id?: number
  courseId: number
  userId?: number
  userName?: string
  content: string
  rating: number
  createTime?: string
  updateTime?: string
}

// 评论统计
export interface CommentStatistics {
  totalComments: number
  averageRating: number
  ratingDistribution: number[] // 5个元素的数组，分别对应1-5星的评论数量
}

// 课程分类
export interface CourseSubject {
  id: number
  name: string
  parentId: number
  sort?: number
  createTime?: string
  updateTime?: string
  children?: CourseSubject[]
}

// 课程浏览记录
export interface CourseView {
  id: number
  userId: number
  courseId: number
  viewTime: string
  ipAddress?: string
  userAgent?: string
  createTime?: string
  viewCount?: number
  // 关联的课程信息
  course?: Course
  // 课程基本信息（扁平化）
  courseTitle?: string
  courseDescription?: string
  courseCover?: string
  coursePrice?: number
  courseViewCount?: number
}

// 批量删除浏览记录请求参数
export interface CourseViewBatchDeleteDTO {
  ids: number[]
}

// 课程搜索参数
export interface CourseSearchParams {
  pageNum?: number
  pageSize?: number
  page?: number
  size?: number
  title?: string
  categoryId?: number
  teacherId?: number
  level?: number
  isFree?: boolean
  minPrice?: number
  maxPrice?: number
  orderBy?: string
}

// 章节
export interface Chapter {
  id: number
  courseId: number
  title: string
  description?: string
  orderNum: number
  status: number
  createTime: string
  updateTime: string
  sections?: Section[]
}

// 小节
export interface Section {
  id: number
  chapterId: number
  courseId: number
  title: string
  content?: string
  videoUrl?: string
  duration?: number
  orderNum: number
  status: number
  createTime: string
  updateTime: string
}

// 课程状态常量
export const CourseStatus = {
  JOINED: 0,    // 已加入
  LEARNING: 1,  // 学习中
  COMPLETED: 2  // 已完成
} as const

// 课程状态标签映射
export const CourseStatusLabels: Record<number, string> = {
  0: '已加入',
  1: '学习中',
  2: '已完成'
}

// 课程状态颜色映射
export const CourseStatusColors: Record<number, string> = {
  0: 'info',      // 已加入 - 蓝色
  1: 'success',   // 学习中 - 绿色
  2: 'warning'    // 已完成 - 橙色
}

File: mqxq/src/main/java/com/jieni/mqxq/domain/entity/Course.java
package com.jieni.mqxq.domain.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

/**
 * 课程实体类
 * 
 * 存储课程的完整信息，包括课程基本信息、价格、状态、统计数据等。
 * 支持课程的分类管理、教师关联、发布状态控制等功能。为在线教育平台的核心业务实体。
 * 
 * @author jieni
 * @version 1.0
 * @since 2025
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Course implements Serializable {
    /**
     * 主键
     */
    @Schema(description = "主键ID")
    private Integer id;
    /**
     * 课程教师ID
     */
    @Schema(description = "课程教师ID")
    private Integer teacherId;
    /**
     * 课程分类ID
     */
    @Schema(description = "课程分类ID")
    private Integer subjectId;
    /**
     * 课程标题
     */
    @Schema(description = "课程标题")
    private String title;
    /**
     * 课程简介
     */
    @Schema(description = "课程简介")
    private String description;
    /**
     * 课程销售价格，设置为0则可免费观看
     */
    @Schema(description = "课程销售价格，设置为0则可免费观看")
    private BigDecimal price;
    /**
     * 总课时
     */
    @Schema(description = "总课时")
    private Integer lessonNum;
    /**
     * 视频总时长（秒）
     */
    @Schema(description = "视频总时长（秒）")
    private Integer durationSum;
    /**
     * 课程封面图片路径
     */
    @Schema(description = "课程封面图片路径")
    private String cover;
    /**
     * 销售数量
     */
    @Schema(description = "销售数量")
    private Integer buyCount;
    /**
     * 浏览数量
     */
    @Schema(description = "浏览数量")
    private Integer viewCount;
    /**
     * 课程状态 0未发布 1已发布
     */
    @Schema(description = "课程状态 0未发布 1已发布")
    private Integer status;
    /**
     * 课程发布时间
     */
    @Schema(description = "课程发布时间")
    private Date publishTime;
    /**
     * 创建时间
     */
    @Schema(description = "创建时间")
    private Date createTime;
    /**
     * 创建者id
     */
    @Schema(description = "创建者id")
    private Integer createUser;
    /**
     * 修改时间
     */
    @Schema(description = "修改时间")
    private Date updateTime;
    /**
     * 修改者id
     */
    @Schema(description = "修改者id")
    private Integer updateUser;
}

